client/src/js/components/bhEmployeeFinder.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * The employee finder component
+ *
+ * @module components/bhEmployeeFinder
+ *
+ * @description a component to deal with employee, it let you choose a employee
+ * 
+ **/
+ 
+//An employee finder component controller
+ 
+function employeeFinderController (employeeService){ 	
+ 	var ctrl = this;
+
+ 	ctrl.searchOptions = [
+ 		{key : 'code', label : 'EMPLOYEE.CODE', placeHolder : 'EMPLOYEE.SEARCH_BY_CODE'},
+ 		{key : 'names', label : 'EMPLOYEE.NAME', placeHolder : 'EMPLOYEE.SEARCH_BY_NAME'}
+ 	];
+
+ 	function load (){
+ 		ctrl.selectedEmployee = null;
+ 		ctrl.employeeValue = null;
+ 		ctrl.session = {state : 'finding'};
+ 		setSearchOption(ctrl.searchOptions[0]); 
+
+ 		employeeService.read()
```

Comment:
This method call downloads all employees.  It would be better to have the typeahead send an $http request and return only 10 results.

 For the cashboxes, we expect to always have less than ten, so this was not necessary.  You can imagine that a client like HealAfrica would have hundreds of employees, so we might as well design for that consideration from the start.